sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java

```diff
@@ -60,10 +60,16 @@ public static String formatDateTimeNullSafe(@Nullable Date date) {
     return date == null ? "" : THREAD_SAFE_DATETIME_FORMAT.format(date);
   }
 
+  @CheckForNull
   public static Date timeToDate(@Nullable Long time) {
     return time == null ? null : new Date(time);
   }
 
+  @CheckForNull
+  public static Long dateToTime(@Nullable Date date) {
+    return date == null ? null : date.getTime();
+  }
+
```

Comment:
IMO method names are misleading because "time" can also be associated to java objects (java.util.Date, ...). I would name them "dateToLong" and "longToDate"